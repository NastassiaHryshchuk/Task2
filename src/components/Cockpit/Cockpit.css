.wrap {
    width: 80%;
    margin: 0 auto;
}

.container {
    padding: 20px 100px;
}

.flex_container {
    display: flex;
}

.results_filter {
    margin-left: auto;
}

.ancore_sub {
    font-weight: bold;
    margin-left: 20px;
    cursor: pointer;
    color: rgb(128, 128, 128);
  }

  .toggle_sub {
    display: none;
  }

  .toggle_sub:checked + label {
    color: rgb(247, 66, 96);
  }