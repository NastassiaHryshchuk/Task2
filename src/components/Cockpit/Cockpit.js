import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { sort } from '../../store/reducer';
import Filter from './Filter/Filter';
import classes from './Cockpit.css';

class Cockpit extends Component {
  constructor(props) {
    super(props);
    this.onSortByReleaseDate = this.onSortByReleaseDate.bind(this);
    this.onSortByRaiting = this.onSortByRaiting.bind(this);
  }

  onSortByReleaseDate(event) {
    event.preventDefault();
    this.props.sort('release_date');
  }

  onSortByRaiting(event) {
    event.preventDefault();
    this.props.sort('vote_average');
  }

  render() {
    return (
      <div className={classes.wrap}>
        <div className={classes.container}>
          <div className={classes.flex_container}>
            <span className={classes.results_count}>{this.props.moves.length} movies found</span>
            <form className={classes.results_filter}>
              <span>Sort by</span>
              <Filter
                onClick={this.onSortByReleaseDate}
                title="release date" />
              <Filter
                onClick={this.onSortByRaiting}
                title="raiting" />
            </form>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    moves: state.films,
  };
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators({ sort }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Cockpit);
