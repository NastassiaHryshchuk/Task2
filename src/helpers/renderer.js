import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Routes from '../Routes/Routes';

export default(req, store) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={req.path} context={{}}>
        <Routes />
      </StaticRouter>
    </Provider>,
  );

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <title>SSR</title>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.INITIAL_STATE = ${JSON.stringify(store.getState())}
        </script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;
};
