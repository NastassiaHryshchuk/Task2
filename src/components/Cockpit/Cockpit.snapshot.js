import React from 'react';
import renderer from 'react-test-renderer';

import Cockpit from './Cockpit';

describe('Cockpit Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <Cockpit />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});