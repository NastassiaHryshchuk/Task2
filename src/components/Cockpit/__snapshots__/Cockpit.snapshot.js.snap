// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cockpit Snapshot renders 1`] = `
<div
  className="wrap"
>
  <div
    className="container"
  >
    <div
      className="flex_container"
    >
      <span
        className="results_count"
      >
        7 movies found
      </span>
      <form
        className="results_filter"
      >
        <span>
          Sort by
        </span>
        <input
          checked={true}
          className="toggle_sub"
          id="releasedateChoice"
          name="film"
          type="radio"
          value="release date"
        />
        <label
          className="ancore_sub"
          htmlFor="releasedateChoice"
        >
          release date
        </label>
        <input
          className="toggle_sub"
          id="raitingChoice"
          name="film"
          type="radio"
          value="raiting"
        />
        <label
          className="ancore_sub"
          htmlFor="raitingChoice"
        >
          raiting
        </label>
      </form>
    </div>
  </div>
</div>
`;
